py/server/deephaven/updateby.py

```diff
@@ -0,0 +1,276 @@
+
+    RESET = _JBadDataBehavior.RESET
+    """Reset the state for the bucket to {@code null} when invalid data is encountered"""
+
+    SKIP = _JBadDataBehavior.SKIP
+    """Skip and do not process the invalid data without changing state"""
+
+    THROW = _JBadDataBehavior.THROW
+    """Throw an exception and abort processing when bad data is encountered"""
+
+    POISON = _JBadDataBehavior.POISON
+    """Allow the bad data to poison the result. This is only valid for use with NaN"""
+
+
+class EmaControl(JObjectWrapper):
+    """A EmaControl represents control parameters for performing EMAs (exponential moving average) with the table
+    update-by operation. """
+    j_object_type = _JEmaControl
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_ema_control
+
+    def __init__(self, on_null: BadDataBehavior = BadDataBehavior.SKIP,
+                 on_nan: BadDataBehavior = BadDataBehavior.SKIP,
+                 on_null_time: BadDataBehavior = BadDataBehavior.SKIP,
+                 on_negative_deltatime: BadDataBehavior = BadDataBehavior.THROW,
+                 on_zero_deltatime: BadDataBehavior = BadDataBehavior.SKIP,
+                 big_value_context: MathContext = MathContext.DECIMAL128):
+        """Initializes a EmaControl for evaluating EMAs (exponential moving average) with table update-by operation.
+
+        Args:
+            on_null (BadDataBehavior): the behavior for when null values are encountered, default is SKIP
+            on_nan (BadDataBehavior): the behavior for when NaN values are encountered, default is SKIP
+            on_null_time (BadDataBehavior): the behavior for when null timestamps are encountered, default is SKIP
+            on_negative_deltatime (BadDataBehavior): the behavior for when negative sample-to-sample time differences
+                are encountered, default is THROW
+            on_zero_deltatime (BadDataBehavior): the behavior for when zero sample-to-sample-time differences are
+                encountered, default is SKIP
+            big_value_context (MathContext): the context to use when processing arbitrary precision numeric values
+                (Java BigDecimal/BigInteger), default is DECIMAL128.
+
+        Raises:
+            DHError
+        """
+        try:
+            j_builder = _JEmaControl.builder()
+            self.j_ema_control = (j_builder.onNullValue(on_null.value)
+                                  .onNanValue(on_nan.value)
+                                  .onNullTime(on_null_time.value)
+                                  .onNegativeDeltaTime(on_negative_deltatime.value)
+                                  .onZeroDeltaTime(on_zero_deltatime.value)
+                                  .bigValueContext(big_value_context.value).build())
+        except Exception as e:
+            raise DHError(e, "failed to build a EmaControl object.") from e
+
+
+class UpdateByClause(JObjectWrapper):
```

Comment:
See rename comment.

Reply:
Created PR #2680 that renamed this object to `OperationControl`